File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map> // 드디어 언오더드 맵을 내가 써보는구나 드디어 !

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string,int> um;
    for (string s : participant) um[s]++;
    for (string s : completion) um[s]--;
    for (auto elem : um) {
        if(elem.second) answer = elem.first; 
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> hm = new HashMap<>();
        
        for(String s : participant) {
            hm.put(s, hm.getOrDefault(s, 0) + 1);
        }
        
        for(String s : completion) {
            if(hm.get(s) == 1) {
                hm.remove(s);
            } else {
                hm.put(s, hm.get(s) - 1);
            }
        }
        
        answer = hm.keySet().iterator().next();
        
        return answer;
    }
}